### DISCUSSION TITLE
Generate Releases and version more frequently

### DISCUSSION THREAD

**Discussion by mfansler** [GreenleafLab/ArchR#2215] [Feature Requests]
I am interested in [packaging ArchR as a Conda package](https://github.com/bioconda/bioconda-recipes/pull/51295), which could help alleviate installation issues. However, it appears `master` has had minor patching since the last Release. Also, `dev` has had many changes and end-users do appear to use this. It would be helpful to capture fixes and features in Releases. Note that one has the ability to mark Releases as `beta`, in case one wishes to signal uncertainty about their stability.

**Comment by immanuelazn** [comment]
Hi Mervin, 

I've seen your recent work with respect to getting a conda package built for ArchR, and am really greatful for that work.  I am currently doing some final checks with making sure that the brunt of the function reference, and vignettes work within my containerized environment. 

Once that is done, I will update 1.0.2 to reflect the current state of master, and 1.0.3 to reflect the current state of dev as a pre-release.

Best,

Immanuel

**Reply by mfansler** [reply]
Excellent news! 🎉 

As you may have seen, we did release a patched version of v1.0.2 on Bioconda. I'll hold off then on a v1.0.3 build until you're released something here. I've already done the background work to get the dependencies added to v1.0.3 available on Conda Forge/Bioconda, so it should be a quick turnaround.

If you have recommendations for what R versions we should try building for, that would be welcome. Currently, we constrained to R 4.1/Bioconductor 3.15, following the advice posted in several Issue responses.

**Reply by immanuelazn** [reply]
Hi Mervin, sorry for the late reply!  Lots of other things coming up on my end and I do apologize.  
1.0.3 will likely be held with respect to the current environment requirements in 1.0.2.  Contradicting the Issue auto-reply, I would recommend building for R 4.1.1 and Bioconductor 3.14 instead as the vignettes were built using them.

I would like to also caveat that ArchR has some heavily suggested dependencies that are not listed within the `DESCRIPTION` file (see this [file](https://github.com/GreenleafLab/ArchR/blob/d9e741c980c7c64e5348c97a74d146cc95f8ba76/R/GlobalDefaults.R#L106)).  I noticed that your conda build does not currently have them listed in your meta.yaml.  Is there any way of getting these packages incorporated as well?

I will add a sub-release to 1.0.3 to support 4.4.1 and bioconductor 3.19 once the deprecated APIs are updated.  

Thanks again for putting in the effort here.  Completely acknowledge it is a mess and we're trying to fix it ASAP on our end too!

**Reply by mfansler** [reply]
> *"...I would recommend building for R 4.1.1 and Bioconductor 3.14 instead as the vignettes were built using them."*

R packages on Bioconda/Conda Forge don't constrain further than `x.x` for R builds, so R 4.1.3 builds will run on all v4.1. Other than R's warning about this, I never seen a concrete report where a package we built had an issue working across R patches. We can adjust the constraints to Bioconductor 3.14.

Personally, I'd like to add some **unit testing**. Rather than trusting that specifying versions is functional, it's better practice to empirically test API calls that are essential. At some point I will try adding something to the Conda recipe, but may also be worth considering adding here, especially as you're refactoring for an R 4.4-compatible version.

> *"...ArchR has some heavily suggested dependencies that are not listed within the DESCRIPTION file..."*

We could do something like issue a `r-archr-full` or `r-archr-recommended` metapackage that installs `r-archr` plus additional suggested dependencies. A major audience for Conda Forge/Bioconda are workflow engineers who prefer minimal install options to run lightweight non-interactive pipelines. For that reason, it's generally better that the minimal install is the default package (`r-archr`) and any bundle(s) with extra features be something users must deliberately specify.

**Reply by immanuelazn** [reply]
Having a metapackage would seem like a good idea once I get this work done!  

Just to be clear, there are a decent chunk of unit tests that are added in 1.0.3, and can be tested using your typical devtools workflow.  I understand it is suboptimal, but currently what I'm doing for ArchR's "test coverage" as of right now is just making sure the vignettes, and the unit tests are able to run.

**Reply by mfansler** [reply]
I see there is a release! 🎉 

For reference, here's [the corresponding Bioconda PR](https://github.com/bioconda/bioconda-recipes/pull/52121). Any suggestion on what R version I should be aiming at? Right now trying R 4.3, which is our latest available.

**Reply by immanuelazn** [reply]
Thanks for being on top of this and sorry for the delayed response!  I was thinking to keeping 1.0.3 to be working with 4.1, and releasing a sub-version with some fixes to dependencies.  One of them being the one you pointed out with ggplot legends.  

On my end, the fixes are working on both 4.3 and 4.4.  I'll ping this discussion board when they're merged in and I create a new release for that.

**Reply by immanuelazn** [reply]
Hi @mfansler , hope you're doing well.  I have updated ArchR to reflect version 1.0.3, and have set the pre-release to point to master with as an official release instead.  Thank you again for your support and suggestions.

**Reply by mfansler** [reply]
I'm a little confused what is going on here. There was a 1.0.3 release in November and that used the v1.0.3 tag. Was that tag changed to point elsewhere?

**Reply by immanuelazn** [reply]
I see, the 1.0.3 was a pre-release and it was pointing to release candidates.  Originally it was connected to a `v1.0.3`, but I changed it to `v1.0.3rc1` and `v1.0.3rc2` while we were still adding in last minute changes.  `v1.0.3` is now a tag that corresponds to the master branch's current commit.

**Reply by puentebravo** [reply]
Thank you all so much for your work in packaging ArchR on Bioconda! I recognize I'm commenting on an old post, but I did see mention of compatibility with R 4.4 mentioned. As far as I'm aware, the current release is only compatible up to R 4.3. 

Are there any plans to make ArchR installable in R 4.4?
